src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,201 @@
+ * \frac{\mu}{\lambda + \mu} + kz \right) \tilde{p}(k)
+ * \f}
+ *
+ * and the strain
+ *
+ * \f{align}
+ * \Theta &= \frac{1}{2 \mu} \int_0^{\infty} dk
+ * J_0(kr) k e^{(-kz)}\left(\frac{-2\mu}{\lambda + \mu}\right) \tilde{p}(k) \\
+ * S_{rr} &= \Theta - S_{\phi\phi} - S_{zz} \\
+ * S_{\phi\phi} &= \frac{\xi_{r}}{r} \\
+ * S_{(rz)} &= -\frac{1}{2 \mu} \int_0^{\infty} dk J_1(kr) k e^{(-kz)}\left(kz
+ * \right) \tilde{p}(k) \\
+ * S_{zz} &= \frac{1}{2 \mu} \int_0^{\infty} dk
+ * J_0(kr) k e^{(-kz)}\left(-\frac{\mu}{\lambda + \mu} - kz \right) \tilde{p}(k)
+ * \f}
+ *
+ * (see Eqs. (11 a) - (11 c) and (13 a) - (13 e), with (13 c) swapped in favor
+ * of (12 c) in \cite Lovelace2007tn), where \f$\tilde{p}(k)= \frac{1}{2\pi}
+ * e^{-(\frac{kr_0}{2})^2}\f$ is the Hankel-Transform of the lasers intensity
+ * profile and \f$ \Theta = \mathrm{Tr}(S)\f$ the materials expansion.
+ *
+ */
+class HalfSpaceMirror {
+ public:
+  static constexpr size_t volume_dim = 3;
+
+  using constitutive_relation_type =
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>;
+
+  struct BeamWidth {
+    using type = double;
+    static constexpr OptionString help{
+        "The lasers beam width r_0 with FWHM = 2*sqrt(ln 2)*r_0"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  struct Material {
+    using type = constitutive_relation_type;
+    static constexpr OptionString help{"The material properties of the beam"};
+  };
+
+  struct IntegrationIntervals {
+    using type = size_t;
+    static constexpr OptionString help{
+        "Workspace size for numerical integrals. Increase if integrals fail to "
+        "reach the prescribed tolerance at large distances relative to the "
+        "beam width. The default values for workspace size and tolerances "
+        "should accommodate distances of up to ~100 beam widths."};
+    static type lower_bound() noexcept { return 1; }
+    static type default_value() noexcept { return 350; }
+  };
+
+  struct AbsoluteTolerance {
+    using type = double;
+    static constexpr OptionString help{
+        "Absolute tolerance for numerical integrals"};
+    static type lower_bound() noexcept { return 0.; }
+    static type default_value() noexcept { return 1e-12; }
+  };
+
+  struct RelativeTolerance {
+    using type = double;
+    static constexpr OptionString help{
+        "Relative tolerance for numerical integrals"};
+    static type lower_bound() noexcept { return 0.; }
+    static type upper_bound() noexcept { return 1.; }
+    static type default_value() noexcept { return 1e-10; }
+  };
+
+  using options = tmpl::list<BeamWidth, Material, IntegrationIntervals,
+                             AbsoluteTolerance, RelativeTolerance>;
+  static constexpr OptionString help{
+      "A semi-infinite mirror on which a laser introduces stress perpendicular "
+      "to the mirrors surface. The displacement then is the Hankel-Transform "
+      "of the general solution multiplied by the beam profile"};
```

Comment:
I don't understand what you mean with "general solution" here. Make it more clear, or just delete the sentence.